src/module_MEDIATOR.F90

```diff
@@ -5474,7 +5605,38 @@ subroutine MedPhase_prep_ocn(gcomp, rc)
     allocate(wgtm01(lbound(icewgt,1):ubound(icewgt,1),lbound(icewgt,2):ubound(icewgt,2)))
     do j=lbound(icewgt,2),ubound(icewgt,2)
     do i=lbound(icewgt,1),ubound(icewgt,1)
-#ifndef FV3_CPLD
+
+!LHC 2020
+#ifdef FV3_CPLD
+      atmwgt(i,j)  = 1.0_ESMF_KIND_R8 - icewgt(i,j)
+      atmwgt1(i,j) = 0.0_ESMF_KIND_R8
+      icewgt1(i,j) =  icewgt(i,j)
+      wgtm01(i,j)  = -atmwgt(i,j)
+      wgtp01(i,j)  =  atmwgt(i,j)
+#elif DATM
+! DATM uses mediator aoflux calc in icy water
+      atmwgt(i,j)  = 1.0_ESMF_KIND_R8 - icewgt(i,j)
+      atmwgt1(i,j) = atmwgt(i,j)
+      icewgt1(i,j) = icewgt(i,j)
+      wgtp01(i,j)  = 0.0_ESMF_KIND_R8
+      wgtm01(i,j)  = 0.0_ESMF_KIND_R8
+! DATM uses mediator aoflux calc in non-icy water
+      if (atmocn_flux_from_atm .and. icewgt(i,j) <= 0.0_ESMF_KIND_R8) then
+        atmwgt1(i,j) = 0.0_ESMF_KIND_R8
+        icewgt1(i,j) = 0.0_ESMF_KIND_R8
+        wgtp01(i,j)  = 1.0_ESMF_KIND_R8
+        wgtm01(i,j)  = -1.0_ESMF_KIND_R8
+      endif
+
+      ! check wgts do add to 1 as expected
+      if (abs(atmwgt(i,j) + icewgt(i,j) - 1.0_ESMF_KIND_R8) > 1.0e-12 .or. &
+          abs(atmwgt1(i,j) + icewgt1(i,j) + wgtp01(i,j) - 1.0_ESMF_KIND_R8) > 1.0e-12 .or. &
+          abs(atmwgt1(i,j) + icewgt1(i,j) - wgtm01(i,j) - 1.0_ESMF_KIND_R8) > 1.0e-12) then
+        call ESMF_LogWrite(trim(subname)//": ERROR atm + ice fracs inconsistent", ESMF_LOGMSG_ERROR, line=__LINE__, file=__FILE__, rc=dbrc)
+        rc = ESMF_FAILURE
+        return
+      endif
+#else
```

Comment:
This is not the right thing to do. The existing code in module_mediators is correct for either using the original flux formulas or the new bulk formulas. You do not want to change the logic here. 

What you do need to do is to create a flag or logical within a namelist which will control which shr_flux_atmOcn is called; either the original or the new _bf version. The flag should default to NOT use the new flux formulas so that the default behaviour of the aofluxes is maintained. 

This may seem like overkill for the NEMS mediator since we are moving the CMEPS, but this will actually be more critical in CMEPS. Other groups and applications use the aofluxes w/in the CMEPS mediator. You must have a way of allowing the default calculation to occur with your formulation only as an option.

Reply:
in the "updated" module_MEDIATOR.F90, the flag of "bulk_method_ori" can control to use either the original or the new bulk method. The default is the original method.